lib/domain/dtos/DtoFactory.js

```diff
@@ -13,36 +13,49 @@
 
 const Joi = require('joi');
 
+const tokenSchema = Joi.object({ token: Joi.string() });
+
 /**
  * Returns a request DTO which require an empty body & params (in-URL parameters) but with a specific list of query parameters
  *
- * @param {object} querySchema the description of query parameters (token is added automatically)
+ * @param {object|Joi.Schema} querySchema the description of query parameters (token is added automatically)
  * @return {object} the resulting DTO
  */
 const queryOnlyDtoFactory = (querySchema) => Joi.object({
-    query: {
-        token: Joi.string(),
-        ...querySchema,
-    },
+    query: tokenSchema.concat(Joi.isSchema(querySchema) ? querySchema : Joi.object(querySchema)),
     body: Joi.object({}),
     params: Joi.object({}),
 });
 
 /**
  * Returns a request DTO which require an empty query & body but with a specific list of params (in-URL parameters)
  *
- * @param {object} paramsSchema the description of params
+ * @param {object|Joi.Schema} paramsSchema the description of params
  * @return {object} the resulting DTO
  */
 const paramsOnlyDtoFactory = (paramsSchema) => Joi.object({
-    query: {
-        token: Joi.string(),
-    },
+    query: tokenSchema,
     body: Joi.object({}),
-    params: paramsSchema,
+    params: Joi.isSchema(paramsSchema) ? paramsSchema : Joi.object(paramsSchema),
 });
 
+/**
+ * Return schema for sorting
+ * @param {string|string[]} names list of names by which query results might be ordered
+ * @returns {Joi.Schema} schema
+ */
+const createSortDto = (names) => {
+    const orderSchema = Joi.string().valid('asc', 'desc', 'ASC', 'DESC');
+    return Array.isArray(names) ?
+        Joi.object(Object.fromEntries(names.map((name) => [name, orderSchema])))
+        : Joi.object({
+            [names]: orderSchema,
+        });
+};
+
 exports.DtoFactory = {
     queryOnly: queryOnlyDtoFactory,
     paramsOnly: paramsOnlyDtoFactory,
+    PaginationDto: require('./PaginationDto.js'),
+    createSortDto,
```

Comment:
done

Reply:
I prefer `sort`, because it feels more natural for me. Ex here https://github.com/AliceO2Group/Bookkeeping/pull/1224/files/15327b3cc44db0fa8494d73e45c32dc608585ee3..d18cab1f904ce49c2a1c907d21c35f375db56c7c#diff-113b24adfbb52acf57ff8983b4e54130f754f7868d2fec79bb0a90e1cecca4dcR37 the property is `sort`

Reply:
..., ok

Reply:
1. I would expect `sort` when executing sorting, what doesn't apply here, but applies for API, because it means imperative `sort`
2. Here we aim for factory producing DTO for checking `sorting` related params, not `sort`
3. `sort` is verb so its inconsistent with other factories, which are named with nouns

Reply:
changed